lib/validator/validator.go

```diff
@@ -26,16 +26,18 @@ import (
 	"github.com/projectcalico/libcalico-go/lib/numorstring"
 	"github.com/projectcalico/libcalico-go/lib/scope"
 	"github.com/projectcalico/libcalico-go/lib/selector"
+	"github.com/projectcalico/libcalico-go/lib/selector/tokenizer"
+
 	"gopkg.in/go-playground/validator.v8"
 )
 
 var validate *validator.Validate
 
 var (
+	labelRegex         = regexp.MustCompile(`^` + tokenizer.LabelKeyMatcher + `?$`)
```

Comment:
Hm. I think the `?` belongs in the LabelKeyMatcher.  Right?

The `?` means that the trailing `([a-zA-Z0-9_.-]{0,61}[a-zA-Z0-9])` can occur once or not at all.